cvmfs/file_processing/chunk.cc

```diff
@@ -57,14 +57,12 @@ void Chunk::Initialize() {
 
   content_hash_context_.buffer = smalloc(content_hash_context_.size);
   shash::Init(content_hash_context_);
-
-  zlib_context_.zalloc   = Z_NULL;
-  zlib_context_.zfree    = Z_NULL;
-  zlib_context_.opaque   = Z_NULL;
-  zlib_context_.next_in  = Z_NULL;
-  zlib_context_.avail_in = 0;
-  const int zlib_retval = deflateInit(&zlib_context_, Z_DEFAULT_COMPRESSION);
-  assert(zlib_retval == 0);
+  
+  if (compression_alg_ == zlib::kZlibDefault) {
+    compressor_ = new zlib::ZlibCompressor();
+  } else if (compression_alg_ == zlib::kNoCompression) {
+    compressor_ = new zlib::EchoCompressor();
+  }
```

Comment:
What about a factory method in the abstract `Compressor` base class to factor out this distinction:
```C++
compressor_ = zlib::Compressor::Construct(compression_alg_);
```
There is even a helper template (`PolymorphicConstruction<>`) in util.h for this.